devenv/main.tf

```diff
@@ -92,19 +88,20 @@ resource "azurerm_key_vault" "keyvault" {
       "Import",
     ]
   }
+}
 
-  access_policy {
-    tenant_id = data.azurerm_client_config.current.tenant_id
-    object_id = data.azurerm_user_assigned_identity.clusteridentity.principal_id
+resource "azurerm_key_vault_access_policy" "allowclusteraccess" {
```

Comment:
moved the keyvault access policy out of the keyvault resource so it can be provisioned concurrently with the aks cluster